tests/conftest.py

```diff
@@ -272,6 +283,32 @@ def to_matlab_type(data):
     else:
         return data  # If the data type is already MATLAB-compatible
     
+# Helper function to compare MATLAB double arrays element-wise, handling NaN comparisons
+def compare_matlab_arrays(result, expected):
+    # Convert MATLAB double to list for comparison
+    result_list = result if isinstance(result, list) else list(result)
+    expected_list = expected if isinstance(expected, list) else list(expected)
+
+    # Check if lengths are the same
+    if len(result_list) != len(expected_list):
+        return False
+    
+    # Compare each element
+    for r, e in zip(result_list, expected_list):
+        # Extract scalar value if wrapped in an array-like structure
+        r_value = r[0] if isinstance(r, (list, np.ndarray, matlab.double)) and len(r) > 0 else r
+        e_value = e[0] if isinstance(e, (list, np.ndarray, matlab.double)) and len(e) > 0 else e
```

Comment:
Possibly. Not for this specific part of the project, but there might be as yet undiscovered functions for which it would be useful to have that feature.

How about for the time being I put in a catch to identify any instances where size>1 is True, and exit? If we need that functionality we can add at that point.

Can also write code for n length now if you think it's necessary. Might just need some recursion or something.

Reply:
fixed with recursion

Reply:
Looks good, marking as resolved.